tests/test_writer.py

```diff
@@ -2480,3 +2480,63 @@ def test_append_pdf_with_dest_without_page(caplog):
     writer.append(reader)
     assert "/__WKANCHOR_8" not in writer.named_destinations
     assert len(writer.named_destinations) == 3
+
+
+def test_writer_contextmanager():
+    """To test the writer with context manager, cf #2912"""
+    pdf_path = str(RESOURCE_ROOT / "crazyones.pdf")
+    with PdfWriter(pdf_path) as w:
+        assert len(w.pages) > 0
+        assert not w.fileobj
+    with open(pdf_path, "rb") as f, PdfWriter(f) as w:
+        assert len(w.pages) > 0
+        assert not w.fileobj
+    with open(pdf_path, "rb") as f, PdfWriter(BytesIO(f.read(-1))) as w:
+        assert len(w.pages) > 0
+        assert not w.fileobj
+
+    try:
+        with NamedTemporaryFile(suffix=".pdf", delete=False) as tmp:
```

Comment:
I would recommend to use the `tmp_path` fixture instead here.

Reply:
the try is normally not required : this is just to be sure to cleanup. i prefer this function to ensure to have a new unused filename

Reply:
But using `def test_writer_contextmanager(tmp_path):`, followed by `tmp_file = tmp_path / "out.pdf"` is much cleaner as it performs the cleanup automatically without having to catch and re-raise any exceptions ourselves.